diff --git a/vendor/magento/module-customer/Block/Adminhtml/Edit/Tab/Newsletter/Grid/Renderer/Action.php b/vendor/magento/module-customer/Block/Adminhtml/Edit/Tab/Newsletter/Grid/Renderer/Action.php
index b9ef3966169..032e2e988df 100644
--- a/vendor/magento/module-customer/Block/Adminhtml/Edit/Tab/Newsletter/Grid/Renderer/Action.php
+++ b/vendor/magento/module-customer/Block/Adminhtml/Edit/Tab/Newsletter/Grid/Renderer/Action.php
@@ -60,7 +60,7 @@ class Action extends \Magento\Backend\Block\Widget\Grid\Column\Renderer\Abstract
                     'newsletter/template/preview',
                     [
                         'id' => $row->getTemplateId(),
-                        'subscriber' => $this->_coreRegistry->registry('subscriber')->getId()
+                        'subscriber' => $row->getSubscriberId()
                     ]
                 ),
                 'target' => '_blank',
diff --git a/vendor/magento/module-newsletter/Model/ResourceModel/Queue/Collection.php b/vendor/magento/module-newsletter/Model/ResourceModel/Queue/Collection.php
index 2914a25ba72..f4e72c61953 100644
--- a/vendor/magento/module-newsletter/Model/ResourceModel/Queue/Collection.php
+++ b/vendor/magento/module-newsletter/Model/ResourceModel/Queue/Collection.php
@@ -225,7 +225,10 @@ class Collection extends \Magento\Framework\Model\ResourceModel\Db\Collection\Ab
             )->join(
                 ['subscriber' => $this->getTable('newsletter_subscriber')],
                 'link.subscriber_id=subscriber.subscriber_id',
-                ['subscriber_store_id' => 'subscriber.store_id']
+                [
+                    'subscriber_store_id' => 'subscriber.store_id',
+                    'subscriber_id' => 'subscriber.subscriber_id',
+                ]
             )->where(
                 'subscriber.customer_id = ?',
                 $customerId
